tools/clang/test/HLSL/ShaderOpArith.xml

```diff
@@ -3482,4 +3482,197 @@ void MSMain(uint GID : SV_GroupIndex,
+      <Descriptor Kind="SAMPLER" AddressU="BORDER" AddressV="BORDER" BorderColorR="1.0" />
+      <Descriptor Kind="SAMPLER" AddressU="BORDER" AddressV="BORDER" BorderColorR="1.0" ComparisonFunc="EQUAL"/>
+    </DescriptorHeap>
+    
+    <DescriptorHeap Name="RtvHeap" NumDescriptors="1" Type="RTV">
+      <Descriptor Name="RTarget" Kind="RTV"/>
+    </DescriptorHeap>
+
+    <InputElements>
+      <InputElement SemanticName="POSITION" Format="R32G32B32_FLOAT" AlignedByteOffset="0" />
+    </InputElements>
+    <RenderTargets>
+      <RenderTarget Name="RTarget">
+        <Viewport Width="2.0" Height="2.0" MaxDepth="1.0"/>
+      </RenderTarget>
+    </RenderTargets>
+
+    <!-- -->
+    <Shader Name="PS66" Target="ps_6_6" EntryPoint="PSMain" Text="@MAIN"/> 
+    <Shader Name="VS66" Target="vs_6_6" EntryPoint="VSMain" Text="@MAIN"/>
+    <Shader Name="CS66" Target="cs_6_6" EntryPoint="main" Text="@MAIN"/> 
+    <!-- -->
+    
+    <!-- FALLBACK BEGINS ->
+    <RootValues>
+      <RootValue HeapName="ResourceDescriptorHeap" />
+    </RootValues>
+    <Shader Name="PS66" Target="ps_6_0" EntryPoint="PSMain" Text="@MAIN" Arguments="/DFALLBACK=1"/>
+    <Shader Name="VS66" Target="vs_6_0" EntryPoint="VSMain" Text="@MAIN" Arguments="/DFALLBACK=1"/>
+    <Shader Name="CS66" Target="cs_6_0" EntryPoint="main" Text="@MAIN" Arguments="/DFALLBACK=1"/>
+    <!- FALLBACK ENDS -->
+
+    <Shader Name="MAIN" Target="cs_6_6" EntryPoint="main">
+      <![CDATA[
+    struct Constants
+    {
+      uint idx0, idx1, idx2, idx3, idx4, idx5, idx6, idx7;
+    };
+
+    #ifdef FALLBACK
+
+    #define ROOT_SIG [RootSignature("RootFlags(ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT), DescriptorTable( SRV(t0), SRV(t1), SRV(t2), UAV(u3), UAV(u4), UAV(u5), CBV(b6), UAV(u7,numDescriptors=3) ), StaticSampler(s0, addressU = TEXTURE_ADDRESS_BORDER, addressV = TEXTURE_ADDRESS_BORDER, borderColor=STATIC_BORDER_COLOR_OPAQUE_WHITE), StaticSampler(s1, addressU = TEXTURE_ADDRESS_BORDER, addressV = TEXTURE_ADDRESS_BORDER, borderColor=STATIC_BORDER_COLOR_OPAQUE_WHITE, comparisonFunc=COMPARISON_EQUAL)")]
+
+    ConstantBuffer<Constants> CB : register(b6);
+
+    ByteAddressBuffer g_rawBuf : register(t0);
+    StructuredBuffer<float> g_structBuf : register(t1);
+    Texture2D<float> g_tex : register(t2);
+    RWByteAddressBuffer g_rwRawBuf : register(u3);
+    RWStructuredBuffer<float> g_rwStructBuf : register(u4);
+    RWTexture1D<float> g_rwTex : register(u5);
+
+    // Result buffers
+    RWBuffer<float> g_result : register(u7);
+    RWBuffer<float> g_resultVS : register(u8);
+    RWBuffer<float> g_resultPS : register(u9);
+
+    SamplerState g_samp : register(s0);
+    SamplerComparisonState g_sampCmp : register(s1);
+
+    #else // NO FALLBACK
+
+    #define ROOT_SIG [RootSignature("RootFlags(CBV_SRV_UAV_HEAP_DIRECTLY_INDEXED | SAMPLER_HEAP_DIRECTLY_INDEXED | ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT), SRV(t0), SRV(t1), SRV(t2), UAV(u3), UAV(u4), UAV(u5), CBV(b6), UAV(u7),  UAV(u8),  UAV(u9)")]
```

Comment:
Why does this have root SRV/UAV arguments in the root signature?  We should get rid of those.